Zig no longer resets horizontal position in `cmd.exe` for compile error messages

### Zig Version

0.13.0-dev.365+332fbb4b0

### Steps to Reproduce and Observed Behavior

Tested on `Microsoft Windows 11 Home`, version `10.0.22000 Build 22000`.
I just updated from `0.13.0-dev.211+6a65561e3` and noticed a difference in compile error output in `cmd.exe`.

```
C:\...>C:\use\Development\zig-windows-x86_64-0.13.0-dev.211+6a65561e3\zig test main.zig
main.zig:27:14: error: expected type 'void', found 'noreturn'
        .a, .b => |p| p, //non-inline shared prong with payload incorrectly triggers "expected type 'void', found 'noreturn'"
            ~^

C:\...>C:\use\Development\zig-windows-x86_64-0.13.0-dev.365+332fbb4b0\zig test main.zig
main.zig:27:14: error: expected type 'void', found 'noreturn'
                                                                     .a, .b => |p| p, //non-inline shared prong with payload incorrectly triggers "expected type 'void', found 'noreturn'"
                                                                              ~^
```

To me it looks like the horizontal position isn't reset; my guess would be that Zig is no longer outputting `\r`, and whatever else it is doing isn't working correctly for my `cmd.exe`.
For the record, on Windows Terminal (which I think is the same as PowerShell?) the output looks correct in both versions.

### Expected Behavior

Correctly resetting the horizontal position on line breaks.

I'm not sure whether this is directly caused by https://github.com/ziglang/zig/pull/20172 - I tried manually reverting [the change in `lib/std/io/tty.zig`](https://github.com/ziglang/zig/pull/20172/files#diff-5ccf43485f42fba787bf81b77c7da82d20ba01ab843bd8d6a7fc4e2d60528baa) , however the output of neither `zig test` nor of the on-demand-compiled `zig fmt` changed.
However, it's also possible that this error displaying logic is pre-compiled in both cases and so expectedly isn't affected by a change in userland `std`.

--- a/lib/std/fs/File.zig
+++ b/lib/std/fs/File.zig
@@ -264,16 +264,17 @@ pub fn getOrEnableAnsiEscapeSupport(self: File) bool {
 
             // For Windows Console, VT Sequences processing support was added in Windows 10 build 14361, but disabled by default.
             // https://devblogs.microsoft.com/commandline/tmux-support-arrives-for-bash-on-ubuntu-on-windows/
-            // Use Microsoft's recommended way to enable virtual terminal processing.
+            //
+            // Note: In Microsoft's example for enabling virtual terminal processing, it
+            // shows attempting to enable `DISABLE_NEWLINE_AUTO_RETURN` as well:
             // https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences#example-of-enabling-virtual-terminal-processing
-            var requested_console_modes: windows.DWORD = windows.ENABLE_VIRTUAL_TERMINAL_PROCESSING | windows.DISABLE_NEWLINE_AUTO_RETURN;
-            var console_mode = original_console_mode | requested_console_modes;
-            if (windows.kernel32.SetConsoleMode(self.handle, console_mode) != 0) return true;
-
-            // An application receiving ERROR_INVALID_PARAMETER with one of the newer console mode
-            // flags in the bit field should gracefully degrade behavior and try again.
-            requested_console_modes = windows.ENABLE_VIRTUAL_TERMINAL_PROCESSING;
-            console_mode = original_console_mode | requested_console_modes;
+            // This is avoided because in the old Windows Console, that flag causes \n (as opposed to \r\n)
+            // to behave unexpectedly (the cursor moves down 1 row but remains on the same column).
+            // Additionally, the default console mode in Windows Terminal does not have
+            // `DISABLE_NEWLINE_AUTO_RETURN` set, so by only enabling `ENABLE_VIRTUAL_TERMINAL_PROCESSING`
+            // we end up matching the mode of Windows Terminal.
+            const requested_console_modes = windows.ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+            const console_mode = original_console_mode | requested_console_modes;
             if (windows.kernel32.SetConsoleMode(self.handle, console_mode) != 0) return true;
         }
 

